runtime/delegates.inc.t4

```diff
@@ -27,10 +27,24 @@ struct Delegates {
 
 static struct Delegates delegates = { 0 };
 
+static guint32
+create_linked_away_exception (const char *func)
+{
+	char *msg = xamarin_strdup_printf ("The runtime function %s has been linked away.", func);
+	guint32 gchandle = xamarin_create_product_exception (8028, msg);
+	xamarin_free (msg);
+	return gchandle;
+}
+
 <# foreach (var d in delegates) { #>
 <#= d.CReturnType #>
 <#= d.EntryPoint #> (<#= d.CArgumentSignature #>)
 {
+	<#if (d.ExceptionHandling && d.OnlyDynamicUsage) {#>if (delegates.<#= d.EntryPoint.Substring ("xamarin_".Length) #> == NULL) {
+		*exception_gchandle = create_linked_away_exception ("<#= d.EntryPoint.Substring ("xamarin_".Length) #>");
+		return<# if (d.CReturnType != "void") { #> (<#= d.CReturnType #>) 0<# } #>;
+	}
+<#}#>
```

Comment:
The T4 code is somewhat hard to read, but the generated function now look like this:

```c
MonoObject *
xamarin_get_block_wrapper_creator (MonoObject * method, int parameter, guint32 *exception_gchandle)
{
	if (delegates.get_block_wrapper_creator == NULL) {
		*exception_gchandle = create_linked_away_exception ("get_block_wrapper_creator");
		return (MonoObject *) 0;
	}
	return delegates.get_block_wrapper_creator (method, parameter, exception_gchandle);
}
```

The `if (delegates.get_block_wrapper_creator == NULL) { ... }` block is the new code.